controlplane/api/v1beta1/rke2controlplane_types.go

```diff
@@ -43,6 +43,18 @@ type RKE2ControlPlaneSpec struct {
 	// Replicas is the number of replicas for the Control Plane.
 	Replicas *int32 `json:"replicas,omitempty"`
 
+	// Version defines the desired Kubernetes version.
+	// This is only a placeholder for now, and the RKE2ConfigSpec.AgentConfig.Version field should be used instead.
+	// In future iterations, this field overrides the RKE2 Version specificied in RKE2ConfigSpec.AgentConfig.Version
+	// which will be deprecated in newer versions of the API.
+	// +optional
+	Version string `json:"version"`
```

Comment:
As this is going to be included in an API version bump, lets change this so the spec.Version is main way to communicate the version. But if its empty and `RKE2ConfigSpec.AgentConfig.Version` is supplied we will use that.

Reply:
Is this version accounted anywhere? In upstream this Version field is slightly unnessesary, as this only may differentiate the control plane machines version from worker machine versions. It might be simpler to unify those. Just thinking outloud, may be wrong

Reply:
This version is a requirement for control plane providers to implement ClusterClass https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/proposals/20210526-cluster-class-and-managed-topologies.md#provider-implementation.

Reply:
Lets follow up after this merges and before we release on the version fields.

Reply:
+1